src/Conf.ml

```diff
@@ -471,6 +472,19 @@ module Formatting = struct
     C.choice ~names:["let-open"] ~all ~doc ~env ~allow_inline:true
       ~update:(fun conf x -> {conf with let_open= x} )
 
+  let line_endings =
+    let doc = "Line endings used." in
+    let all =
+      [ ( "preserve"
+        , `Preserve
+        , "$(b,preserve) preserves the original line endings." )
+      ; ("unix", `Unix, "$(b,unix) uses Unix line endings.")
+      ; ("windows", `Windows, "$(b,windows) uses Windows line endings.") ]
+    in
+    let env = Arg.env_var "OCAMLFORMAT_LINE_ENDINGS" in
+    C.choice ~names:["line-endings"] ~all ~doc ~env ~allow_inline:true
```

Comment:
Ok, `~allow_inline:true` isn't (always) going to work, since the output file is opened before the input attributes are read and processed. This seems ok, just less convenient for us updating test results I suppose.